simulator/modules/fetch/bpu/t/unit_test.cpp

```diff
@@ -226,16 +226,16 @@ TEST_CASE( "Cache Miss")
 
     // Check default cache miss behaviour
     Addr PC = 12;
-    CHECK_FALSE(bp->is_taken(PC));
+    CHECK_FALSE(bp->is_hit(PC));
 
     PC = 16;
-    CHECK_FALSE(bp->is_taken(PC));
+    CHECK_FALSE(bp->is_hit(PC));
 
     PC = 20;
-    CHECK_FALSE(bp->is_taken(PC));
+    CHECK_FALSE(bp->is_hit(PC));
 
     PC = 12;
-    CHECK_FALSE(bp->is_taken(PC));
+    CHECK_FALSE(bp->is_hit(PC));
```

Comment:
Could you please test that `is_taken` is also false? In current implementation `is_taken` is implicated by `is_hit`, but that may change for always-hit or always-miss branch predictions).